ludwig/schema/metadata/configs/llm.yaml

```diff
@@ -0,0 +1,98 @@
+generation:
+  temperature:
+    ui_display_name: Temperature
+    default_value_reasoning:
+      Increasing the temperature will allow the model to generate more diverse sequences,
+      but will also increase the likelihood of generating nonsense. As such, we recommend setting this value to
+      something closer to 0 for classification tasks, and something closer to 1 for text generation tasks where the
+      goal is to generate novel text.
+    expected_impact: 3
+  max_new_tokens:
+    ui_display_name: Max New Tokens
+    default_value_reasoning:
+      Increasing the maximum number of new tokens will allow the model
+      to generate longer sequences, but because inference time scales linearly with the sequence length,
+      longer sequences will be much slower to generate. For classification tasks, it's generally better to
+      use a smaller number of new tokens, while for text generation tasks, it's generally better to use a larger
+      number of new tokens.
+    expected_impact: 3
+  num_beams:
+    ui_display_name: Number of Beams
+    default_value_reasoning:
+      Increasing the number of beams will allow the model to generate more diverse sequences,
+      but will also increase inference time. Some backends (like DeepSpeed) also do not support beam search.
+      As such, we recommend leaving this as 1 in most cases, unless you're finding the quality of the generated
+      sequences to be lacking.
+    expected_impact: 2
+  top_k:
+    ui_display_name: Top K
+    expected_impact: 2
+  top_p:
+    ui_display_name: Top P
+    expected_impact: 2
+  max_length:
+    ui_display_name: Max Length
+    expected_impact: 2
+  min_length:
+    ui_display_name: Min Length
+    expected_impact: 2
+  min_new_tokens:
+    ui_display_name: Min New Tokens
+    expected_impact: 2
+  do_sample:
+    ui_display_name: Do Sample
+    expected_impact: 2
+  use_cache:
+    ui_display_name: Use Cache
+    expected_impact: 2
+prompt:
+  retrieval:
+    type:
+      ui_display_name: Type
+      expected_impact: 3
+    index_name:
+      ui_display_name: Index Name
+      expected_impact: 2
+    model_name:
+      ui_display_name: Model Name
+      expected_impact: 2
+    k:
+      ui_display_name: Top K
+      expected_impact: 2
+  task:
+    ui_display_name: Task
+    expected_impact: 3
+    component_type: textarea
+  template:
+    ui_display_name: Template
+    expected_impact: 3
+    component_type: textarea
```

Comment:
So based on a separate conversation with @martindavis and I had earlier today, I would have expected the new `component_type` property to appear at the _schema_ level, not in the parameter metadata.

By schema level I mean in the `json_schema_type_mapping` for `PromptConfigField`. So it would become a "sibling" property of "parameter_metadata" rather than one of the latter's child props.

We can plumb it through either way, but by putting it in parameter metadata the correct thing to do would be to define a default for this prop in the `ParameterMetadata` class as well, which I suppose could be `null`. But doing so would also imply we're establishing a general standard of tying parameters to their renderable HTML elements.

Up to you guys, I'm fine going either way. I kind of lean in this direction since we are already putting other "UI" focused metadata in their like the "ui_display_name". But maybe we don't to more strongly tie (quite light) business logic in like this.

Reply:
Thanks! I went ahead and added `ui_component_type` to `ParameterMetadata`, which I feel is consistent with how we handle the display name, etc.